src/traces/scattergl/index.js

```diff
@@ -439,10 +439,17 @@ function plot(gd, subplot, cdata) {
                 var pos = [], srcPos = (lineOptions && lineOptions.positions) || stash.positions;
 
                 if(trace.fill === 'tozeroy') {
-                    pos = [srcPos[0], 0];
-                    pos = pos.concat(srcPos);
-                    pos.push(srcPos[srcPos.length - 2]);
-                    pos.push(0);
+                    var firstdef = 1;
+                    while(isNaN(srcPos[firstdef])) {
+                        firstdef += 2;
+                    }
+                    var lastdef = srcPos.length - 1;
+                    while(isNaN(srcPos[lastdef])) {
+                        lastdef += -2;
+                    }
+                    pos = [ srcPos[firstdef - 1], 0 ];
+                    pos = pos.concat(srcPos.slice(firstdef - 1, lastdef + 1));
+                    pos = pos.concat([ srcPos[lastdef - 1], 0 ]);
```

Comment:
I notice the existing code didn't take advantage of this, but `push` is variadic:
`pos.push(srcPos[lastdef - 1], 0);`